kasa/tests/test_dimmer.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+
+from .conftest import dimmer
+
+
+@dimmer
+async def test_set_brightness(dev, mocker):
+    query_helper = mocker.patch("kasa.SmartDimmer._query_helper")
+    await dev.set_brightness(99)
+
+    query_helper.assert_called_with(
+        "smartlife.iot.dimmer", "set_brightness", {"brightness": 99},
+    )
+
+
+@dimmer
+async def test_set_brightness_transition(dev, mocker):
+    query_helper = mocker.patch("kasa.SmartDimmer._query_helper")
+    await dev.set_brightness(99, transition=1000)
+
+    query_helper.assert_called_with(
+        "smartlife.iot.dimmer",
+        "set_dimmer_transition",
+        {"brightness": 99, "duration": 1000},
+    )
```

Comment:
It would be great to have also tests for the implicit turn on / turn off when using the transition.